packages/backend/src/services/UserService.ts

```diff
@@ -320,6 +326,17 @@ export class UserService {
             openIdUser.openId.subject,
         );
 
+        if (
```

Comment:
This won't trigger for a newly registered user. This should execute at the bottom of this function with the userUuid from all three cases:
- User is logged in with existing openid
- User is linked to a new openid 
- User is registered with a new openid

Reply:
Thanks for spotting this! I've added this check + insert on those 3 cases. LMK if you would like me to go further in refactoring: https://github.com/lightdash/lightdash/pull/8508/commits/b0e1e46858816e2c2019af91faf3cce0f1b57e2f